### ISSUE TITLE
how to build 1-dim input to N-dim output LSTM-RNN 

### ISSUE THREAD

**Issue by gamehere007** [fchollet/keras#2836]
If I have a input tensor shape as:

(input_examples,time_steps,input_dim)

and a output tensor shape as:

(output_examples,time_steps,output_dim)

, where input_dim = 1, output_dim = 2. How do I build up the LSTM-RNN architecture? 

Thanks a lot!

**Comment by gamehere007** [comment]
The following code is what I use for my LSTM-RNN architecture.
`hidden = 32
# set model
model = Sequential()
model.add(LSTM(input_dim = 1, 
                output_dim = hidden,
                return_sequences = True))
model.add(Dropout(0.2)) 
model.add(TimeDistributedDense(output_dim = 2))
model.compile(loss='mse', optimizer='rmsprop')`

The code can be ran. But the performance(score) is not good. Any ideas to revise this architecture to improve the performance?


Many thanks,
Oscar

**Comment by xingdi-eric-yuan** [comment]
try to use word embedding

**Comment by gamehere007** [comment]
> try to use word embedding

My goal is to do time series prediction. Predicting two pieces of  time series with the information of another piece of time series

**Comment by jskDr** [comment]
Hi Eric.
Could you let me know what is 'Word Embedding'? Is it supported in Keras?

Best,


On Fri, May 27, 2016 at 2:46 PM Xingdi (Eric) Yuan <notifications@github.com>
wrote:

> try to use word embedding
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fchollet/keras/issues/2836#issuecomment-222224376>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACchna8BqAN0JBB_KZuPR6fFu9MCAfbQks5qFzwBgaJpZM4IorLR>
> .
>
-- 

Sungjin (James) Kim, PhD

Postdoc, CCB in Harvard

sungjinkim@fas.harvard.edu

[Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/

[Linkedin] https://www.linkedin.com/in/jamessungjinkim

[Facebook] https://www.facebook.com/jamessungjin.kim

[alternative email] jamessungjin.kim@gmail.com
